### PR TITLE
digital ian cartridge game

### PR DESCRIPTION
**Pull request by laok233**
## About the PR

added digital ian cartridge game :tm:  (it's cool i guess)
note: i've also planned a hunger system, but it is delayed (lack of time and skill issue) and it is a week a way just like surgery.

**Media**

https://github.com/space-wizards/space-station-14/assets/72604018/1df36725-4e1a-49a0-b9bb-29b95d00a194

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: end
- add: Nanotrasen has rewarded HoP a game cartridge in honor of his contribution to the station.

### CODE REVIEW

**Comment by Equivocateur** [comment]
Ideal coping mechanism when the actual corgo got sent to the chef for free bicaridine

File: Content.Client/CartridgeLoader/Cartridges/DigitalIanUi.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        _fragment = new DigitalIanUiFragment();
+        _fragment.OnFeedButtonPressed += () => SendDigitalIanMessage(DigitalIanUiAction.Feed, userInterface);
+        _fragment.OnPetButtonPressed += () => SendDigitalIanMessage(DigitalIanUiAction.Pet, userInterface);
+    }
+
+    public override void UpdateState(BoundUserInterfaceState state)
+    {
+        // No state to update
+    }
+    private void SendDigitalIanMessage(DigitalIanUiAction action, BoundUserInterface userInterface)
```
Comment (0x6273) [inline Content.Client/CartridgeLoader/Cartridges/DigitalIanUi.cs:29]:
```suggestion
    }
    
    private void SendDigitalIanMessage(DigitalIanUiAction action, BoundUserInterface userInterface)
```

File: Content.Server/CartridgeLoader/Cartridges/DigitalIanCartridgeComponent.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+[RegisterComponent]
+public sealed class DigitalIanCartridgeComponent : Component
+{
+    [DataField("soundFeed")]
+    public SoundSpecifier SoundFeed = new SoundPathSpecifier("/Audio/Items/eatfood.ogg");
+
+    [DataField("soundpet")]
+    public SoundSpecifier SoundPet = new SoundPathSpecifier("/Audio/Animals/small_dog_bark_happy.ogg");
+
+}
```
Comment (0x6273) [inline Content.Server/CartridgeLoader/Cartridges/DigitalIanCartridgeComponent.cs:14]:
```suggestion
    [DataField("soundPet")]
    public SoundSpecifier SoundPet = new SoundPathSpecifier("/Audio/Animals/small_dog_bark_happy.ogg");
}
```

**Review by 0x6273 — CHANGES_REQUESTED** [review]
You need to fix your editor so it does not do this:
image

File: Content.Server/CartridgeLoader/Cartridges/DigitalIanCartridgeSystem.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
+using Content.Shared.CartridgeLoader;
+using Content.Shared.CartridgeLoader.Cartridges;
+using Content.Shared.Popups;
+
+namespace Content.Server.CartridgeLoader.Cartridges;
+public sealed class DigitalIanCartridgeSystem : EntitySystem
```
Comment (0x6273) [inline Content.Server/CartridgeLoader/Cartridges/DigitalIanCartridgeSystem.cs:6]:
```suggestion
namespace Content.Server.CartridgeLoader.Cartridges;

public sealed class DigitalIanCartridgeSystem : EntitySystem
```

**Comment by Kromabae** [comment]
ian tamagotchi moments

**Comment by laok233** [comment]
> You need to fix your editor so it does not do this: image

What does that mean?
No trailing blank line?
And how do i fix it.

**Comment by 0x6273** [comment]
> What does that mean? No trailing blank line? And how do i fix it.

It means that the last line doesn't have a linebreak (\n) character at the end. Most editors should fix this automatically.

File: Content.Client/CartridgeLoader/Cartridges/DigitalIanUiFragment.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    public event Action? OnFeedButtonPressed;
+    public event Action? OnPetButtonPressed;
+
+    public DigitalIanUiFragment()
+    {
+        RobustXamlLoader.Load(this);
+        FeedButton.OnPressed += _ => OnFeedButtonPressed?.Invoke();
+        PetButton.OnPressed += _ => OnPetButtonPressed?.Invoke();
+
+        Icon.SetFromSpriteSpecifier(new SpriteSpecifier.Rsi(new("Mobs/Pets/corgi.rsi"), "ian"));
```
Comment (AJCM-git) [inline Content.Client/CartridgeLoader/Cartridges/DigitalIanUiFragment.xaml.cs:20]: unhardcode this, make it a EntityPrototype datafield in the component

File: Content.Client/CartridgeLoader/Cartridges/DigitalIanUiFragment.xaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    <!-- <BoxContainer  Orientation="Vertical" HorizontalExpand="True" VerticalExpand="True">
+        <Label Text="Hunger" StyleClasses="StatusFieldTitle" />
+        <ProgressBar Name="HungerBar" MaxValue="100">
+            lol <Label Name="HungerText" Margin="4 0" Text="0 %" /> 
+        </ProgressBar>
+        <Label Text="Fun" StyleClasses="StatusFieldTitle" />
+        <ProgressBar Name="FunBar" MaxValue="100">
+            <Label Name="FunText" Margin="4 0" Text="0 %" />
+        </ProgressBar>
+    </BoxContainer> -->
```
Comment (AJCM-git) [inline Content.Client/CartridgeLoader/Cartridges/DigitalIanUiFragment.xaml:23]: This is way too simple, at least add hunger (could just add HungerComponent to the embedded entity and then make a control showcasing the hunger), also rebrand mentions of "Ian" to "Mascot" to make this more generic

**Review by AJCM-git — CHANGES_REQUESTED** [review]

**Comment by laok233** [comment]
well i guess it is better to close this pr and reopen it when i have time to do things.

**Pull request closed without merging**